tests/lava/test_utils/test_float2fixed.py

```diff
@@ -0,0 +1,1553 @@
+from lava.proc.dense.process import Dense
+from lava.magma.compiler import compiler_graphs
+from lava.magma.core.run_configs import Loihi1SimCfg
+from collections import OrderedDict
+
+
+class Proc(AbstractProcess):
+    """A dummy proc with two Vars, one dynamic, one constant."""
+
+    def __init__(self, shape=(1,), **kwargs):
+        u = kwargs.get('u', 0)
+        v = kwargs.get('v', 5)
+        super().__init__(**kwargs)
+        self.u = Var(shape=shape, init=u)
+        self.v = Var(shape=shape, init=v)
+        self.inport = InPort(shape=shape)
+        self.outport = OutPort(shape=shape)
+        self.shape = shape
+
+
+class ProcDense(AbstractProcess):
+    """A dummy dense process with only a weight."""
+
+    def __init__(self, **kwargs):
+        w = kwargs.get('w', 0)
+        super().__init__(**kwargs)
+        self.w = Var(shape=(1,), init=w)
+        self.outport = OutPort(shape=(1,))
+        self.inport = InPort(shape=(1,))
+
+
+class HierProc(AbstractProcess):
+    """A dummy hierarchical Process with In- and OutPorts. Will wrap around a
+    Proc Process."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.inport = InPort(shape=(1,))
+        self.outport = OutPort(shape=(1,))
+
+
+class ProcRef(AbstractProcess):
+    """A dummy process with RefPort."""
+    def __init__(self, **kwargs):
+        u = kwargs.get('u', 0)
+        v = kwargs.get('v', 5)
+        w = kwargs.get('w', 5)
+        super().__init__(**kwargs)
+        self.refport = RefPort(shape=(1,))
+        self.v = Var(shape=(1,), init=v)
+        self.u = Var(shape=(1,), init=u)
+        self.w = Var(shape=(1,), init=w)
+
+
+class ProcVar(AbstractProcess):
+    """A dummy process to which RefProc connects.."""
+    def __init__(self, **kwargs):
+        v = kwargs.get('v', 5)
+        super().__init__(**kwargs)
+        self.v = Var(shape=(1,), init=v)
+
+
+@implements(proc=Proc, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt_pytype_target_err')
+class ProcPyProcModelTargetErr(PyLoihiProcessModel):
+    """A minimal fixed-point PyProcModel implementing Proc."""
+    u: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=17,
+                               domain=None, constant=True, num_bits_exp=4,
+                               meta_parameter=False)
+    v: np.ndarray = LavaPyType(np.ndarray, np.int32, precision="u:24:0",
+                               domain=None, constant=False,
+                               num_bits_exp=None, meta_parameter=False)
+    inport: PyInPort = LavaPyType(PyInPort.VEC_DENSE, np.int16,
+                                  precision="u:16:3")
+    outport: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, np.int16,
+                                    precision="u:16:0")
+
+
+@implements(proc=Proc, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt_pytype_var_err')
+class ProcPyProcModelVarErr(PyLoihiProcessModel):
+    """A minimal fixed-point PyProcModel implementing Proc."""
```

Comment:
The docstrings should mention inhowfar the different PyProcModels differ mutually.